src/Reflex/Dom/SemanticUI/Dropdown.hs

```diff
@@ -283,3 +269,168 @@ semUiDropdownWithItems elId opts iv vals attrs = do
   pb <- getPostBuild
   performEvent_ (liftJSM (activateSemUiDropdownEl $ _element_raw elDD) <$ pb)
   holdDyn iv (elChoice)
+
+------------------------------------------------------------------------------
+-- New implementation
+
+#ifdef ghcjs_HOST_OS
+activateDropdown :: DOM.Element -> (Text -> JSM ()) -> JSM ()
+activateDropdown e onChange = do
+  cb <- asyncCallback1 $ onChange . pFromJSVal
+  js_activateDropdown e cb
+foreign import javascript unsafe
+  "$($1).dropdown({ forceSelection : false \
+                  , onChange: function(value){ $2(value); } \
+                  , fullTextSearch: true });"
+  js_activateDropdown :: DOM.Element -> Callback (JSVal -> JSM ()) -> JSM ()
+#else
+activateDropdown :: DOM.Element -> (Text -> JSM ()) -> JSM ()
+activateDropdown e onChange = do
+  o <- obj
+  o ^. jss ("forceSelection"::Text) False
+  o ^. jss ("onChange"::Text) (fun $ \_ _ [t, _, _] ->
+    onChange =<< fromJSValUnchecked t)
+  o ^. jss ("fullTextSearch"::Text) True
+  void $ jsg1 ("$"::Text) e ^. js1 ("dropdown"::Text) o
+#endif
+
+#ifdef ghcjs_HOST_OS
+dropdownSetExactly :: DOM.Element -> [Int] -> JSM ()
+dropdownSetExactly e is = do
+  jsVal <- toJSVal $ map (tshow) is
+  js_dropdownSetExactly e jsVal
+
+foreign import javascript unsafe
+  "$($1).dropdown('set exactly', $2);"
+  js_dropdownSetExactly :: DOM.Element -> JSVal -> JSM ()
+#else
+dropdownSetExactly :: DOM.Element -> [Int] -> JSM ()
+dropdownSetExactly e is =
+  void $ jsg1 ("$"::Text) e ^. js2 ("dropdown"::Text) ("set exactly"::Text) is
+#endif
+
+------------------------------------------------------------------------------
+
+-- | Config for new semantic dropdowns
+data DropdownConf t a = DropdownConf
+  { _dropdownConf_setValue :: Dynamic t a
+  , _dropdownConf_attributes :: Map Text Text
+  , _dropdownConf_placeholder :: Text
+  }
+
+$(makeLenses ''DropdownConf)
+
+instance (Reflex t) => Default (DropdownConf t (Maybe a)) where
+  def = DropdownConf
+    { _dropdownConf_setValue = constDyn Nothing
+    , _dropdownConf_attributes = mempty
+    , _dropdownConf_placeholder = mempty
+    }
+
+instance (Reflex t) => Default (DropdownConf t [a]) where
+  def = DropdownConf
+    { _dropdownConf_setValue = constDyn mempty
+    , _dropdownConf_attributes = mempty
+    , _dropdownConf_placeholder = mempty
+    }
+
+instance HasAttributes (DropdownConf t a) where
+  type Attrs (DropdownConf t a) = Map Text Text
+  attributes = dropdownConf_attributes
+
+instance HasSetValue (DropdownConf t a) where
+  type SetValue (DropdownConf t a) = Dynamic t a
+  setValue = dropdownConf_setValue
+
+-- | Helper function
+indexToItem :: Vector (a, DropdownItemConfig m) -> Text -> Maybe a
+indexToItem items i' = do
+  i <- readMaybe $ T.unpack i'
+  fst <$> items !? i
+
+-- | Semantic-UI dropdown with static items
+semUiDropdownNew
+  :: (MonadWidget t m, Eq a)
+  => Vector (a, DropdownItemConfig m) -- ^ Items
```

Comment:
I don't see much value in using `Vector` over something much simpler like `[]`.  Instead of depending on `Vector`, I'd rather depend on something that gets us more value.  My recommendation would be the non-empty-zipper package.  http://hackage.haskell.org/package/non-empty-zipper-0.1.0.7/docs/Data-List-NonEmptyZipper.html  It allows you to guarantee that the selected item is in the list.